web/app/components/comment/comment.jsx

```diff
@@ -335,10 +340,50 @@ export default class Comment extends Component {
     });
   }
 
+  isAdmin() {
+    return !this.state.guest && store.get('user').admin;
+  }
+
+  isGuest() {
+    return this.state.guest || !Object.keys(store.get('user')).length;
+  }
+
+  isCurrentUser() {
+    return (
+      (this.props.data && this.props.data.user && this.props.data.user.id) ===
+      (store.get('user') && store.get('user').id)
+    );
+  }
+
+  /**
+   * returns reason for disabled downvote
+   *
+   * @return {(string|null)}
+   */
+  getDownvoteDisabledReason() {
```

Comment:
Is there a reason for two identical methods?

Reply:
Aleksei, thanks for asking! I doubt too, but my intent was that disabling upvote and downvote reasons may differ, mainly for downvote, for example it you want to limit downvote or disable it at all, or want to disable downvoting admins, or mark some controversial comment as non-downvotable, etc..., so I tried to think ahead of time. Maybe there is no reason for this now, and I'm overcomplicating it.

Reply:
Added commit, as an alternative.